server/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java

```diff
@@ -498,7 +516,9 @@ public boolean timeExceeded() {
     }
 
     public void throwTimeExceededException() {
-        throw new TimeExceededException();
+        if (timeoutOverwrites.getOrDefault(Thread.currentThread(), false) == false) {
```

Comment:
Looking at the changes above I am wondering if there are situations where post collection does want timeout to be thrown. Are there? If not is there a way to disable timeouts in post collection directly? I get worried that this type of change will make it harder to migrate to lucene's timeout support.

Reply:
could we not doing something similar to what we were doing before? I mean, what is the point of having timeouts if we don't throw an exception when there is one? Should we rather remove the timeout runnable at this point before post collection? Or are you worried that we may not honour cancellation if we do so?

Reply:
>what is the point of having timeouts if we don't throw an exception when there is one?

Reply:
Note that before we also just build the top level internal aggregations when timeout occurred (in `AggregationPhase`). This workaround allows us to do the same, otherwise we can't return partial aggregation response (we just fail producing the search response).

Reply:
Ok got it, thanks for explaining.